ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -125,7 +126,13 @@ func (r *RayClusterReconciler) Reconcile(ctx context.Context, request ctrl.Reque
 		}
 		return ctrl.Result{RequeueAfter: DefaultRequeueDuration}, err
 	}
-	if err := r.reconcileServices(instance); err != nil {
+	if err := r.reconcileServices(instance, common.HeadService); err != nil {
+		if updateErr := r.updateClusterState(instance, rayiov1alpha1.Failed); updateErr != nil {
+			log.Error(updateErr, "RayCluster update state error", "cluster name", request.Name)
+		}
+		return ctrl.Result{RequeueAfter: DefaultRequeueDuration}, err
+	}
+	if err := r.reconcileServices(instance, common.AgentService); err != nil {
```

Comment:
Should we guard this behind the flag in the annotations too?
i.e. do not create the agent service if the flag is not set to "true"?

Reply:
sounds good